### PR TITLE
Update Dockerfiles to use non-slim base OS images per new guidelines

### PR DESCRIPTION
**Pull request by cjbj**
See https://github.com/oracle/docker-images/blob/main/CONTRIBUTING.md#base-image-rules

Signed-off-by: Christopher Jones <christopher.jones@oracle.com>

### CODE REVIEW

**Comment by Djelibeybi** [comment]
The mandate not to use `-slim` is only for Oracle Linux 8 (because of `microdnf` and lack support for AppStreams/modularity). You do not (and should not) change images based on `oraclelinux:7-slim`.

File: OracleInstantClient/oraclelinux7/12.2.0.1/Dockerfile [hunk 1]
```diff
@@ -18,10 +18,11 @@
...
 # Put all downloaded files in the same directory as this Dockerfile
 # Run: 
-#      $ docker build --pull -t oracle/instantclient:12.2.0.1 .
+#      $ docker build --pull -t oraclelinux7-instantclient:12.2 .
+#      $ docker run -ti --rm oraclelinux7-instantclient:12.2 sqlplus -l /nolog
 #
-#
-FROM oraclelinux:7-slim
+
+FROM oraclelinux:7
```
Comment (Djelibeybi) [inline OracleInstantClient/oraclelinux7/12.2.0.1/Dockerfile:25]: This should stay as `7-slim`

File: OracleInstantClient/oraclelinux7/18/Dockerfile [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
 #
 # Run:
-#      $ docker build --pull -t oracle/instantclient:18 .
+#      $ docker build --pull -t oraclelinux7-instantclient:18 .
+#      $ docker run -ti --rm oraclelinux7-instantclient:18 sqlplus -l /nolog
 #
-#
-FROM oraclelinux:7-slim
+
+FROM oraclelinux:7
```
Comment (Djelibeybi) [inline OracleInstantClient/oraclelinux7/18/Dockerfile:15]: This should stay as `7-slim`

File: OracleInstantClient/oraclelinux7/21/Dockerfile [hunk 3]
```diff
@@ -69,7 +69,7 @@
 # and can be pulled with:
 #   docker pull ghcr.io/oracle/oraclelinux7-instantclient:21
 
-FROM oraclelinux:7-slim
+FROM oraclelinux:7
```
Comment (Djelibeybi) [inline OracleInstantClient/oraclelinux7/21/Dockerfile:72]: This should stay as `7-slim`

File: OracleInstantClient/oraclelinux7/19/Dockerfile [hunk 4]
```diff
@@ -69,7 +69,7 @@
 # and can be pulled with:
 #   docker pull ghcr.io/oracle/oraclelinux7-instantclient:19
 
-FROM oraclelinux:7-slim
+FROM oraclelinux:7
```
Comment (Djelibeybi) [inline OracleInstantClient/oraclelinux7/19/Dockerfile:72]: This should stay as `7-slim`

**Review by Djelibeybi — CHANGES_REQUESTED** [review]

**Comment by cjbj** [comment]
I'll submit a clean PR

**Pull request closed without merging**